### PR TITLE
Test for zypper single transaction

### PR DESCRIPTION
**Pull request by asdil12**
See: https://lists.opensuse.org/archives/list/zypp-devel@lists.opensuse.org/thread/YEFSZZY7FRGJPGPWKNCUT3UXQWVENOCL/
Ticket: https://progress.opensuse.org/issues/105948

### CODE REVIEW

**Comment by asdil12** [comment]
vr:
sle: http://kazhua.qa.suse.de/tests/66#step/zypper_in/24

**Comment by punkioudi** [comment]
One quick thing that I see is that you include only the VRs from 15-sp1. Is this test scheduled only on 15-sp1, or also in other sle maintainace versions?
If it runs also in all the other sle versions, include the VRs for these cases too :)

**Comment by asdil12** [comment]
http://kazhua.qa.suse.de/tests/89#step/zypper_in/10

**Comment by punkioudi** [comment]
> I guess this should be enough to show that old releases are not affected by this PR: http://kazhua.qa.suse.de/tests/89#step/zypper_in/10

yes, indeed

**Review by punkioudi — APPROVED** [review]

**Comment by DimStar77** [comment]
https://openqa.opensuse.org/tests/overview?version=Staging%3AA&distri=opensuse&groupid=2&build=A.327.1

This seems so to be caused by this test installing packages and not cleaning up .. calling, or rather shouting _revert me_

Blocking staging

**Comment by asdil12** [comment]
@DimStar77 https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14448

**Comment by DimStar77** [comment]
> @DimStar77 #14448

Failed tests from staging a restarted ... Thanks for the quick fix

**Comment by DimStar77** [comment]
more fallouts:

https://openqa.opensuse.org/tests/2235306#step/zypper_in/26 (s390x)

**Comment by asdil12** [comment]
> more fallouts:
> 
> https://openqa.opensuse.org/tests/2235306#step/zypper_in/26 (s390x)

I'm debugging that right now but it looks like a product bug to me so far.

**Comment by asdil12** [comment]
--> https://bugzilla.suse.com/show_bug.cgi?id=1196925

File: tests/console/zypper_in.pm [hunk 1]
```diff
@@ -32,6 +46,32 @@ sub run {
...
+        # and also this ZYPP_SINGLE_RPMTRANS feature flag is only available as of 15.4
+        assert_script_run 'touch /preinstall_fail';
+        my $r = script_run 'zypper -n in --allow-unsigned-rpm ' . data_url('zypper/hello0.rpm');
+        die "Unexpected zypper exit code $r - expected 8" unless (defined($r) && ($r == 8));
+        assert_script_run 'rm -f /preinstall_fail';
+
+        assert_script_run 'export ZYPP_SINGLE_RPMTRANS=1';
+
+        assert_script_run 'touch /preinstall_fail';
+        $r = script_run 'zypper -n in --allow-unsigned-rpm ' . data_url('zypper/hello0.rpm');
```
Comment (ge0r) [inline tests/console/zypper_in.pm:62]: Hey Dominik, I believe this line is failing on offline medium runs.
For instance in https://openqa.suse.de/tests/8364256#step/zypper_in/12 (and the other archs)

Comment (asdil12) [reply tests/console/zypper_in.pm:62]: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14557

**Pull request merged**